File: ios/Skoverlay.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Skoverlay, NSObject)

RCT_EXTERN_METHOD(showOverlay:(string)a withB:(string)b withC:(string)c
                  withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/Skoverlay.swift
@objc(Skoverlay)
class Skoverlay: NSObject {

  @objc(showOverlay:withB:withResolver:withRejecter:)
  func showOverlay(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
    resolve(a*b)
  }
}
